app/main/views/register.py

```diff
@@ -150,7 +153,20 @@ def set_up_your_profile():
         if code and state:
             access_token = sign_in._get_access_token(code, state)
             user_email, user_uuid = sign_in._get_user_email_and_uuid(access_token)
-            redirect_url = request.args.get("next")
+
+            invite_data = state.encode("utf8")
+            invite_data = base64.b64decode(invite_data)
+            invite_data = json.loads(invite_data)
+            invited_service = Service.from_id(invite_data["service_id"])
+            invited_user_id = invite_data["invited_user_id"]
+            invited_user = InvitedUser.by_id(invited_user_id)
```

Comment:
This theoretically shouldn't be a situation we can encounter, but just in case, the original code was also checking that the invited user's email matches the email of the user attempting to login:

```python
if user.is_active:
    if session.get("invited_user_id"):
        invited_user = InvitedUser.from_session()
        if user.email_address.lower() != invited_user.email_address.lower():
            flash("You cannot accept an invite for another person.")
            session.pop("invited_user_id", None)
            abort(403)
        else:
            invited_user.accept_invite()
```

We get `user_email` now from Login.gov just a bit above, so a similar check code be done now, right?

```suggestion
            invited_user = InvitedUser.by_id(invited_user_id)
            
            if user_email.lower() != invited_user.email_address.lower():
                flash("You cannot accept an invite for another person.")
                session.pop("invited_user_id", None)
                abort(403)
            else:
                invited_user.accept_invite()
```